pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -141,24 +141,44 @@ func (s *akvKMProvider) GetCertificates(ctx context.Context) (map[keymanagementp
 		logger.GetLogger(ctx, logOpt).Debugf("fetching secret from key vault, certName %v,  keyvault %v", keyVaultCert.Name, s.vaultURI)
 
 		// fetch the object from Key Vault
-		// GetSecret is required so we can fetch the entire cert chain. See issue https://github.com/ratify-project/ratify/issues/695 for details
-		startTime := time.Now()
-		secretBundle, err := s.kvClient.GetSecret(ctx, s.vaultURI, keyVaultCert.Name, keyVaultCert.Version)
+		certBundle, err := s.kvClient.GetCertificate(ctx, s.vaultURI, keyVaultCert.Name, keyVaultCert.Version)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to get secret objectName:%s, objectVersion:%s, error: %w", keyVaultCert.Name, keyVaultCert.Version, err)
+			return nil, nil, fmt.Errorf("failed to get certificate objectName:%s, objectVersion:%s, error: %w", keyVaultCert.Name, keyVaultCert.Version, err)
 		}
 
-		certResult, certProperty, err := getCertsFromSecretBundle(ctx, secretBundle, keyVaultCert.Name)
-		if err != nil {
-			return nil, nil, fmt.Errorf("failed to get certificates from secret bundle:%w", err)
-		}
+		//TODO: can I replace else with a continue?
+		if !*certBundle.Attributes.Enabled {
+			fmt.Printf("debug: certificate %s version %s is disabled.", keyVaultCert.Name, keyVaultCert.Version)
+
+			isEnabled := "false"
+			startTime := time.Now()
+			lastRefreshed := startTime.Format(time.RFC3339)
+
+			metrics.ReportAKVCertificateDuration(ctx, time.Since(startTime).Milliseconds(), keyVaultCert.Name)
+			certProperty := getStatusProperty(keyVaultCert.Name, keyVaultCert.Version, isEnabled, lastRefreshed)
+			certsStatus = append(certsStatus, certProperty)
+			certMapKey := keymanagementprovider.KMPMapKey{Name: keyVaultCert.Name, Version: keyVaultCert.Version, Enabled: isEnabled}
+			certsMap[certMapKey] = []*x509.Certificate{} // empty cert chain
```

Comment:
where is the certificate value (without chain) set in `certsMap`?

Reply:
When the certificate is disabled, the certificate isn't stored at all. My thinking was that this would prevent the `FlattenKMPMap` function from including any disabled certificates. Self-signed certs for example could be used to verify if that cert was passed to the verifier because it doesn't have any cert chain beyond itself. That said, now that there is an `enabled` field it's possible to update the Flatten function to look for only enabled certs.

QQ: by certificate value, you're meaning the cert in x509 format?

Reply:
I didn't realize this was for the disabled case. My bad. I should've looked at the if block condition above. Can you remind me if we plan to show if a certificate is disabled at all in status? I'm trying to understand if the operation on L165 is to clear map value but still have the key available to signify it is disabled.

Reply:
Not a problem. :) And yes, the status is now shown in the KMP status. Here's an example:

```bash
 Certificates:
      Enabled:         true
      Last Refreshed:  2024-10-21T20:27:23Z
      Name:            ratify
      Version:         0ff373a9259c4578a247cfd7861a8804
```
L165 is meant to clear the value so the disabled certificate data so it's not stored in the KMP but the key remains. The key values are then used to populate the status. 

Technically, speaking I don't think there is any reason the key need to remain in the map either, other than to match the status that's being reflected to the user.

Reply:
After looking into the way we use certsMap, seems we can just remove L165 since the disabled cert will not be used anywhere. Correct me if I'm wrong.

Reply:
You're absolutely correct, the zero value will be the same. Making line 165 redundant.

Reply:
@akashsinghal what do you think about not storing disabled cert entries in the KMP cache? Then if a cert is found to be disabled it's reported in the status but removed from the cache.

Reply:
Sure that should work too because the status will always get updated alongside an upstream KV request fetch

Reply:
In order to set this up there's several changes that'll need to be made, should that be in this PR or a separate PR?

Reply:
@akashsinghal @binbin-li I got this working in a separate [branch](https://github.com/duffney/ratify/tree/issue-1752/rm-disabled-certs) by adding a few functions to the KMP to delete entries from a resource in cache. This takes care of the edge case where there is only one certificate or key stored in the cache that was once enabled and is now disabled. I can either merge it into this branch or submit it as PR once this one makes it through. Lmk which is prefered. :)

Reply:
I would prefer merging into this PR as we already have context on it.

Reply:
Changes merged into this PR. :)